triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Converter.scala

```diff
@@ -34,205 +36,133 @@ import com.daml.platform.participant.util.LfEngineToApi.{
 import scala.concurrent.duration.{FiniteDuration, MICROSECONDS}
 
 // Convert from a Ledger API transaction to an SValue corresponding to a Message from the Daml.Trigger module
-case class Converter(
-    fromTransaction: Transaction => Either[String, SValue],
-    fromCompletion: Completion => Either[String, SValue],
-    fromHeartbeat: SValue,
-    fromACS: Seq[CreatedEvent] => Either[String, SValue],
-    toFiniteDuration: SValue => Either[String, FiniteDuration],
-    toCommands: SValue => Either[String, Seq[Command]],
-    toRegisteredTemplates: SValue => Either[String, Seq[Identifier]],
-)
+final class Converter(compiledPackages: CompiledPackages, triggerIds: TriggerIds) {
```

Comment:
I'd always thought this was an odd way to define the converter - use a case class and instantiate via an apply!